caravan.lua

```diff
@@ -1,31 +1,311 @@
+            handle_ctrl_click_on_render = true
+        end
+
+        if handle_ctrl_click_on_render or handle_shift_click_on_render then
+            copyGoodflagState()
+        end
+    end
+    CaravanOverlay.super.onInput(self, keys)
+end
+
+function CaravanOverlay:render(dc)
+    if not is_trade_overlay_active() then return false end
+    if handle_shift_click_on_render then
+        handle_shift_click_on_render = false
+        select_shift_clicked_container_items(df.global.game.main_interface.trade.goodflag[0], trader_selected_state, 0)
+        select_shift_clicked_container_items(df.global.game.main_interface.trade.goodflag[1], broker_selected_state, 1)
+    elseif handle_ctrl_click_on_render then
+        handle_ctrl_click_on_render = false
+        collapse_ctrl_clicked_containers(df.global.game.main_interface.trade.goodflag[0], trader_selected_state, 0)
+        collapse_ctrl_clicked_containers(df.global.game.main_interface.trade.goodflag[1], broker_selected_state, 1)
+    end
+
+    CaravanOverlay.super.render(self, dc)
+end
+
+
+CaravanInfoOverlay = defclass(CaravanInfoOverlay, overlay.OverlayWidget)
+CaravanInfoOverlay.ATTRS{
+    default_pos={x=-3,y=-12},
+    default_enabled=true,
+    viewscreens='dwarfmode',
+    frame={w=27, h=13},
+    frame_style=gui.MEDIUM_FRAME,
+    frame_background=gui.CLEAR_PEN,
+}
+
+function CaravanInfoOverlay:init()
+    self:addviews{
+        widgets.Label{
+            frame={t=0, l=0},
+            text = 'Shift+Click checkbox:',
+            text_pen=COLOR_LIGHTGREEN,
+        },
+        widgets.Label{
+            frame={t=1, l=0},
+            text = 'select items inside bin',
+        },
+        widgets.Label{
+            frame={t=3, l=0},
+            text = 'Ctrl+Click checkbox:',
+            text_pen=COLOR_LIGHTGREEN,
+        },
+        widgets.Label{
+            frame={t=4, l=0},
+            text = 'collapse single bin',
+        },
+        widgets.HotkeyLabel{
+            frame={t=6, l=0},
+            label='collapse bins',
+            key='CUSTOM_CTRL_C',
+            on_activate=collapseAllContainers,
+        },
+        widgets.HotkeyLabel{
+            frame={t=7, l=0},
+            label='collapse all',
+            key='CUSTOM_CTRL_X',
+            on_activate=collapseEverything,
+        },
+        widgets.Label{
+            frame={t=9, l=0},
+            text = 'Shift+Scroll:',
+            text_pen=COLOR_LIGHTGREEN,
+        },
+        widgets.Label{
+            frame={t=9, l=14},
+            text = 'fast scroll',
+        },
+        CaravanOverlay{},
+    }
+end
+
+function CaravanInfoOverlay:render(dc)
+    if not is_trade_overlay_active() then return false end
+    CaravanInfoOverlay.super.render(self, dc)
+end
+
+function CaravanInfoOverlay:onInput(keys)
+    if not is_trade_overlay_active() then return false end
+    CaravanInfoOverlay.super.onInput(self, keys)
+end
+
+OVERLAY_WIDGETS = {
+    overlay=CaravanInfoOverlay,
```

Comment:
this will show as "caravan.overlay" which maybe isn't *quite* descriptive enough. How about `tradeScreenExtension` or *something* that explicitly refers to the trade screen.